tests/test_client.py

```diff
@@ -0,0 +1,184 @@
+import pytest
+import testinfra
+import subprocess
+import os
+import time
+import json
+
+
+@pytest.fixture
+def entities():
+    return {
+        'organization_label': 'Test_Organization',
+        'activation_key': 'Test AK',
+        'product': 'Test Product',
+        'product_label': 'Test_Product',
+        'yum_repository': 'Zoo',
+        'yum_repository_label': 'Zoo',
+        'errata_id': 'RHEA-2012:0055',
+        'container_repository_label': 'foremanbusybox'
+    }
+
+@pytest.fixture(scope='session')
+def client(request):
+    name = "smoker-client-%s" % request.param.replace('/', '_').replace(':', '_')
+    podman_id = subprocess.check_output(['podman', 'ps', '-q', '--filter', "name=%s" % name])
+
+    if not podman_id:
+        subprocess.check_output(['podman', 'run', '--name', name, '-dt', request.param]).decode().strip()
+
+    # return a testinfra connection to the container
+    yield testinfra.get_host("podman://" + name)
+
+@pytest.fixture(scope='session')
+def smoker():
+    return testinfra.get_host("local://localhost")
+
+@pytest.fixture
+def server_hostname(variables):
+    assert variables['server_hostname']
+    return variables['server_hostname']
+
+@pytest.fixture
+def foreman_version(variables):
+    assert variables['foreman_version']
+    return variables['foreman_version']
+
+def pytest_generate_tests(metafunc):
+    with open(metafunc.config.option.variables[0]) as file:
+      variables = json.load(file)
+      metafunc.parametrize('client', variables['clients'], indirect=True, scope='session')
+
+def teardown_module():
+    podman_ids = subprocess.check_output(['podman', 'ps', '-q', '--filter', "name=smoker-client"]).decode().strip()
+    podman_ids = podman_ids.split("\n")
+
+    for id in podman_ids:
+        subprocess.check_call(['podman', 'rm', '-f', id])
+
+
+@pytest.mark.katello_agent
+def test_install_subscription_manager(client, smoker):
+    client.run('yum -y install subscription-manager')
+    sub_man = client.package('subscription-manager')
+    assert sub_man.is_installed
+
+    hammer = smoker.package('tfm-rubygem-hammer_cli_katello')
+    assert hammer.is_installed
+
+
+@pytest.mark.katello_agent
+def test_register_subscription_manager_with_username_and_password(client, entities, server_hostname):
+    client.run("yum erase -y 'katello-ca-consumer-*'")
+
+    client.run_expect([0], "rpm -Uvh http://%s/pub/katello-ca-consumer-latest.noarch.rpm" % server_hostname)
```

Comment:
I'd introduce a fixture for the consumer rpm url. If we have multiple, we can also use `pytest_generate_tests` to create a matrix.